meilisearch-http/src/routes/search.rs

```diff
@@ -25,38 +27,39 @@ struct SearchQuery {
     crop_length: Option<usize>,
     attributes_to_highlight: Option<String>,
     filters: Option<String>,
-    timeout_ms: Option<u64>,
     matches: Option<bool>,
 }
 
-pub async fn search_with_url_query(ctx: Request<Data>) -> SResult<Response> {
-    ctx.is_allowed(Public)?;
+#[get("/indexes/{index_uid}/search", wrap = "Authentication::Public")]
+async fn search_with_url_query(
+    data: web::Data<Data>,
+    path: web::Path<IndexParam>,
+    params: web::Query<SearchQuery>,
+) -> Result<HttpResponse, ResponseError> {
+    let index = data
+        .db
+        .open_index(&path.index_uid)
+        .ok_or(ResponseError::index_not_found(&path.index_uid))?;
 
-    let index = ctx.index()?;
-    let db = &ctx.state().db;
-    let reader = db.main_read_txn()?;
+    let reader = data.db.main_read_txn()?;
 
     let schema = index
         .main
         .schema(&reader)?
-        .ok_or(ResponseError::open_index("No Schema found"))?;
+        .ok_or(ResponseError::internal("Impossible to retrieve the schema"))?;
 
-    let query: SearchQuery = ctx
-        .query()
-        .map_err(|_| ResponseError::bad_request("invalid query parameter"))?;
+    let mut search_builder = index.new_search(params.q.clone());
```

Comment:
I am wondering about this clone, is it necessary?

Reply:
Yes, I think. new_search take a String. I could change it but to do that I need to change the SearchBuilder.